### PR TITLE
Implement pressure altitude and density altitude for full 86km of ISA

### PR DESCRIPTION
**Pull request by seanmcleod**
@djlinse you'll be happy to know that the troposphere specific check based on altitude is now gone and there are no troposphere specific variables anymore. 

I've added pressure altitude and density altitude checks to the Python test cases to test for at least 1 altitude in every ISA defined layer up to 86km.

### CODE REVIEW

**Comment by seanmcleod** [comment]
@djlinse good questions as usual. I've generally followed the existing patterns in the existing code and tried to minimize my code changes. I'm not sure why the original implementation implemented the atmosphere by using pre-calculated temps instead of storing the ISA defined lapse rates instead.

**Comment by seanmcleod** [comment]
Busy looking into the tests that are failing, none of which are the atmosphere tests.

**Comment by bcoconni** [comment]
@seanmcleod 
> I'm not sure why the original implementation implemented the atmosphere by using pre-calculated temps instead of storing the ISA defined lapse rates instead.

My guess is that temperatures are stored in an instance of `FGTable` because it provides the linear interpolation between the breakpoints with a simple call
```c++
StdAtmosTemperatureTable.GetValue(GeoPotAlt)
```
whereas we have to explicitly code the interpolation for pressure because it is not linear and `FGTable` provides nothing else than linear interpolation.

Another point is that lapse rates in SI unit need very few digit precision whereas the same lapse rates in British units need much more precision. So one could consider that it is simpler to compute the lapse rates in British units rather than explicitly provide them.

And even if you would provide ISA lapse rates as pre-calculated data, you would need to store the temperature breakpoints as we are doing for pressure. So you would not save a data structure.

Just my 2 cents.

File: src/models/atmosphere/FGStandardAtmosphere.cpp [hunk 1]
```diff
@@ -445,32 +439,82 @@ void FGStandardAtmosphere::ResetSLPressure()
...
-double FGStandardAtmosphere::CalculateDensityAltitude(double density, double geoalt)
+void FGStandardAtmosphere::CalculateStdDensityBreakpoints()
 {
-  if (geoalt > TroposphereMaxAltitude)
-    return geoalt;
-  else {
-    // Convert to density altitude based on ratio of density to standard sea-level density.
-    // Passed in density has taken into account any non standard ISA day changes.
-    double density_altitude = TroposphereAltitudeScaleFactor * (1.0 - pow(density / StdSLdensity, DATroposphereExponent));
+  StdDensityBreakpointVector.resize(StdPressureBreakpointVector.size());
```
[the remark this answers is not in the document]
Comment (bcoconni) [reply src/models/atmosphere/FGStandardAtmosphere.cpp]: This one is optional :smiley:

**Comment by djlinse** [comment]
@bcoconni I see why the FGStandard uses the temperature table and understand why it might have been developed in that fashion originally.  But this is one of the very few places in coding of a complicated physical model when you have an exact standard that says what the number should be.  Just my OCD-ness, I guess.

I hadn't really thought about/noticed how the standard temperature table was being used/abused as far as the 0/1 indexing.  Not a good coding convention, but possibly/probably not worth worrying about.

@seanmcleod I still will would suggest that rather than calculate Lmb in each routine, that you make a variable StdLapseRate (or StdLapseRateVector if you must), and assign it a copy of LapseRateVector in InitModel, much like I did with StdPressureBreakpointVector, as at that point LapseRateVector is exactly what you are calculating over and over again, I believe.

**Pull request merged**